Python tests failing with nanobind changes

### Report

It looks like the tests can't find the existing `libOpenMS.so` file:

```
ImportError: libOpenMS.so: cannot open shared object file: No such file or directory
ImportError while importing test module '/build/source/build/pyOpenMS/tests/unittests/test_spectrum_zerocopy.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/3cgrdksdb6l2ifbp0ag5h94zk8mldib6-python3-3.13.12-env/lib/python3.13/importlib/__init__.py:88: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
tests/unittests/test_spectrum_zerocopy.py:4: in <module>
    import pyopenms as poms
pyopenms/__init__.py:236: in <module>
    _import_submodules()
pyopenms/__init__.py:229: in _import_submodules
    raise ImportError(
E   ImportError: Failed to import any pyOpenMS bindings.
E   Make sure the package was built correctly with nanobind.
```

The file `build/lib/libOpenMS.so` exists and is valid.  Any idea why the tests can't find it?

--- a/src/pyOpenMS/CMakeLists.txt
+++ b/src/pyOpenMS/CMakeLists.txt
@@ -235,7 +235,6 @@ else()
     if(NOT PYOPENMS_PREPARE_WHEEL_REPAIR)
       set_target_properties(${target_name} PROPERTIES
         INSTALL_RPATH "${PY_RPATH}"
-        BUILD_WITH_INSTALL_RPATH 1
       )
       if(APPLE)
         set_target_properties(${target_name} PROPERTIES

--- a/src/pyOpenMS/CMakeLists.txt
+++ b/src/pyOpenMS/CMakeLists.txt
@@ -236,6 +236,13 @@ else()
       set_target_properties(${target_name} PROPERTIES
         INSTALL_RPATH "${PY_RPATH}"
       )
+      if(NOT NO_DEPENDENCIES)
+        # When dependencies are bundled alongside the modules, use the install
+        # RPATH at build time so $ORIGIN/ finds the co-located libraries.
+        set_target_properties(${target_name} PROPERTIES
+          BUILD_WITH_INSTALL_RPATH 1
+        )
+      endif()
       if(APPLE)
         set_target_properties(${target_name} PROPERTIES
           BUILD_WITH_INSTALL_NAME_DIR 1

